app/controllers/unsubscriptions_controller.rb

```diff
@@ -28,13 +36,36 @@ def confirmed
 
   def set_attributes
     @id = params.require(:id)
-    @subscription = api.get_subscription(@id)
-    @title = @subscription.dig("subscription", "subscriber_list", "title").presence
-    @authenticated_for_subscription = check_authenticated(@subscription)
+    @original_subscription = api.get_subscription(@id)
+    @latest_subscription = api.get_latest_matching_subscription(@id)
+    @title = @latest_subscription.dig("subscription", "subscriber_list", "title").presence
+    @authenticated_for_subscription = check_authenticated(@latest_subscription)
+  end
+
+  def subscription_is_latest
```

Comment:
For this method, `old_subscription_has_same_frequency_as_latest` and `already_unsubscribed`, I'd suggest adding a question mark to the end of the name, since a boolean is returned

Reply:
Love it! Keep forgetting I can do that.